sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/LogRecordDataAssert.java

```diff
@@ -160,6 +166,107 @@ public LogRecordDataAssert hasBody(String body) {
     return this;
   }
 
+  @SuppressWarnings({"MethodCanBeStatic"})
+  private void bodyIsAnyValue() {
+    // Actually, we can't do these two checks because the body is still STRING type
+    //    assertThat(actual.getBody().getType()).isNotSameAs(Body.Type.EMPTY);
+    //    assertThat(actual.getBody().getType()).isNotSameAs(Body.Type.STRING);
```

Comment:
I think we can make this assertion now.. See how [LogMarshaler](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/LogMarshaler.java#L65-L76) checks if `body instanceof AnyValueBody`.

I might be misunderstanding though. Definitely need to be careful to not expose AnyValue as part of the public API until its stable.

Reply:
I think its time we stabilized the AnyValue portions of the log API. Opened #6581 to track.

Its strange to have stable APIs asserting the contents of a log record body which are impossible to produce with using an experimental API. As a part of stabilizing we should fix the `Body.Type` enum, either by including all the possible value types of AnyValue OR by extending it with a single AnyValue type.